app/src/helpers/windowHelpers.ts

```diff
@@ -218,33 +213,52 @@ export function injectCSS(browserWindow: BrowserWindow): void {
         details: OnHeadersReceivedListenerDetails,
         callback: (headersReceivedResponse: HeadersReceivedResponse) => void,
       ) => {
-        log.debug(
-          'browserWindow.webContents.session.webRequest.onHeadersReceived',
-          { details, callback },
-        );
-        if (details.webContents) {
-          details.webContents
-            .insertCSS(cssToInject)
-            .catch((err) => {
-              log.error('webContents.insertCSS ERROR', err);
-            })
-            .finally(() =>
-              callback({
-                cancel: false,
-                responseHeaders: details.responseHeaders,
-              }),
-            );
-        } else {
-          callback({
-            cancel: false,
-            responseHeaders: details.responseHeaders,
+        injectCSSIntoResponse(details, cssToInject)
+          .then((responseHeaders) => {
+            callback({
+              cancel: false,
+              responseHeaders,
+            });
+          })
+          .catch((err) => {
+            log.error('injectCSSIntoResponse ERROR', err);
+            callback({
+              cancel: false,
+              responseHeaders: details.responseHeaders,
+            });
           });
-        }
       },
     );
   });
 }
 
+async function injectCSSIntoResponse(
+  details: OnHeadersReceivedListenerDetails,
+  cssToInject: string,
+): Promise<Record<string, string[]>> {
+  const nonInjectableMethods = ['DELETE', 'OPTIONS'];
```

Comment:
Your PR description rationale would be useful here:

```suggestion
  // We go with a denylist rather than a whitelist (e.g. only GET/html)
  // to avoid "whoops I didn't think this should have been CSS-injected" cases
  const nonInjectableMethods = ['DELETE', 'OPTIONS'];
```